tests/unit/test_value.c

```diff
@@ -248,7 +253,29 @@ SENTRY_TEST(value_json_parsing)
     sentry_value_decref(rv);
 
     rv = sentry__value_from_json(
-        STRING("[42, \"foo\\u2603\", \"bar\", {\"foo\": 42}]"));
-    TEST_CHECK_JSON_VALUE(rv, "[42,\"foo☃\",\"bar\",{\"foo\":42}]");
+        STRING("[false, 42, \"foo\\u2603\", \"bar\", {\"foo\": 42}]"));
+    TEST_CHECK_JSON_VALUE(rv, "[false,42,\"foo☃\",\"bar\",{\"foo\":42}]");
+    sentry_value_decref(rv);
+
+    rv = sentry__value_from_json(
+        STRING("{\"escapes\": "
+               "\"quot: \\\", backslash: \\\\, slash: \\/, backspace: \\b, "
+               "formfeed: \\f, linefeed: \\n, carriage: \\r, tab: \\t\", "
+               "\"surrogates\": "
+               "\"\\uD801\\udc37\"}"));
+    // escaped forward slashes are parsed, but not generated
+    TEST_CHECK_JSON_VALUE(rv,
+        "{\"escapes\":"
+        "\"quot: \\\", backslash: \\\\, slash: /, backspace: \\b, "
+        "formfeed: \\f, linefeed: \\n, carriage: \\r, tab: \\t\","
+        "\"surrogates\":\"𐐷\"}");
+    sentry_value_decref(rv);
+
+    // unmatched surrogates don’t parse
+    rv = sentry__value_from_json(STRING("\"\\uD801\""));
+    TEST_CHECK(sentry_value_is_null(rv));
+    rv = sentry__value_from_json(
+        STRING("{\"valid key\": true, \"invalid key \\uD801\": false}"));
+    TEST_CHECK_JSON_VALUE(rv, "{\"valid key\":true}");
```

Comment:
_nit_: I'd prefer to have these in separate tests with a low number of checks. If one of them fails, it's easier to know what regressed (independently).